src/Symfony/Component/Serializer/Serialize/Mapping/PropertyMetadata.php

```diff
@@ -0,0 +1,108 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Serializer\Serialize\Mapping;
+
+use Symfony\Component\Serializer\Exception\InvalidArgumentException;
+use Symfony\Component\Serializer\Type\Type;
+
+/**
+ * Holds serialization metadata about a given property.
+ *
+ * @author Mathias Arlaud <mathias.arlaud@gmail.com>
+ *
+ * @experimental in 7.0
```

Comment:
I'd set this as `@internal`

Reply:
By setting this class to internal, it'll be impossible to decorate the `PropertyMetadataLoaderInterface` which is according to me a really powerful extension point.

As I've written in the PR description:
> This extension point is kind of easy to use and powerful.
> It can be compared to the way normalizers are used nowadays in terms of "data content"

Therefore, I'd keep that public instead, WDYT?